src/manage.c

```diff
@@ -497,9 +497,9 @@ static void manage_track()
 
     acq_status_t *acq = &acq_status[sid_to_index(ch->sid)];
 
-    /* Is ephemeris marked unhealthy? */
+    /* Is ephemeris or alert flag marked unhealthy?*/
     const ephemeris_t *e = ephemeris_get(ch->sid);
-    if (e->valid && !e->healthy) {
+    if (ch->nav_msg.alert || (e->valid && !e->healthy)) {
```

Comment:
This alert flag should probably also be passed through and checked before using a sat in a solution.  This function is called periodically to drop bad sats, but it is independent of the solution code.